pythonforandroid/recipes/android/__init__.py

```diff
@@ -51,6 +51,9 @@ def prebuild_arch(self, arch):
             'PY2': 0,
             'JAVA_NAMESPACE': java_ns,
             'JNI_NAMESPACE': jni_ns,
+            'ACTIVITY_CLASS_NAME': self.ctx.activity_class_name,
+            'ACTIVITY_PACKAGE': self.ctx.activity_package,
+            'ACTIVITY_NAMESPACE': self.ctx.activity_package.replace('.', '/'),
```

Comment:
How about introducing a `ACTIVITY_FULL_CLASS_NAME` value that is already concatenating both `activity_class_name` and `activity_package`?
e.g.
```python
{
    "ACTIVITY_FULL_CLASS_NAME": f"{self.ctx.activity_package}.{self.ctx.activity_class_name}",
}
```
That way you don't need to carry over the concatenation all over later in the code

Reply:
👍  i will check that!

Reply:
ok... now i just added one input parameter `activity_class_name` which must be a full name of the Java class.